osu.Game/Collections/DrawableCollectionListItem.cs

```diff
@@ -163,19 +173,41 @@ private void load(OsuColour colours)
                     // but that subscription does not only cover *changes to the set of collections* (i.e. addition/removal/rearrangement of collections),
                     // but also covers *changes to the properties of collections*, which `BeatmapMD5Hashes` is one.
                     // when a collection item changes due to `BeatmapMD5Hashes` changing, the list item is deleted and re-inserted, thus guaranteeing this to work correctly.
-                    int count = collection.PerformRead(c => c.BeatmapMD5Hashes.Count);
+                    updateCountText();
 
-                    countText.Text = count == 1
-                        // Intentionally not localised until we have proper support for this (see https://github.com/ppy/osu-framework/pull/4918
-                        // but also in this case we want support for formatting a number within a string).
-                        ? $"{count:#,0} item"
-                        : $"{count:#,0} items";
+                    // subscribe to all BeatmapInfo changes
+                    subscription = realmAccess.RegisterForNotifications(
+                        realm => realm.All<BeatmapInfo>(),
+                        (_, __) => Schedule(updateCountText)
+                    );
                 }
                 else
                 {
                     PlaceholderText = "Create a new collection";
                 }
             }
+
+            private void updateCountText()
+            {
+                List<string> beatmapHashes = collection.PerformRead(c => c.BeatmapMD5Hashes.ToList());
+                int total = beatmapHashes.Count;
+                int missing = beatmapHashes.Count(hash => beatmapManager.QueryBeatmap(b => b.MD5Hash == hash) == null);
```

Comment:
So let me get this straight:

`.Count()` here is LINQ. Meaning that for every beatmap hash in a collection, this is going to perform one beatmap lookup by hash. This happens every time any beatmap in the game is added, modified or removed. And also this is going to simultaneously be queried for every single collection existing in the game when the N subscriptions for N collections fire. 

I haven't ran this yet, but let's just say I'm skeptical this will be anywhere near acceptable performance. Benchmarking on a not-small database would be very appreciated here.